lib/next_ls/code_actionable.ex

```diff
@@ -0,0 +1,19 @@
+defmodule NextLS.CodeActionable do
+  # A diagnostic can produce 1 or more code actions, also it would be easier for
+  # unsupported diagnostics to return an empty list when gathering the code actions
+  # from diagnostics
+
+  defmodule Data do
+    defstruct [:diagnostic, :uri, :document]
+  end
+
+  @callback to_code_action(arg :: Data.t()) :: [CodeAction.t()]
+
+  def from(:elixir, diagnostic_data) do
```

Comment:
These will need to be dispatched to the extension processes dynamically, as eventually libraries can provide their own extensions and we won't statically know what extensions exist or have their code loaded. 

This is fine for now,  it maybe leave a todo comment

Reply:
Maybe we can store the module name to dispatch to in the data attribute of the diagnostic? And libraries will have to obey the contract for this to work.
I think we can do something like:
```
def from(:elixir, data), do: NextLS.ElixirExtension.CodeAction.from(data)
def from(:credo, data), do: NextLS.CredoExtension.CodeAction.from(data)
def from(namespace, data), do: Module.concat(namespace, CodeAction).from(data)
```
Then it's going to be clear for external libraries what behaviour they need to implement.

I still like the static part being there since it's probably faster for the BEAM

Reply:
The code might not even be loaded on the NextLS node, which is by dispatching it to the extension process will be required.

Reply:
Ah I see, then I'll leave it as a TODO comment.